src/hivpy/hiv_testing.py

```diff
@@ -145,67 +194,77 @@ def update_anc_hiv_testing(self, pop, time_step):
-            pop.set_present_variable(col.EVER_TESTED, True, just_tested)
-            pop.set_present_variable(col.LAST_TEST_DATE, pop.date, just_tested)
-            # "reset" dummy partner columns
-            pop.set_present_variable(col.NSTP_LAST_TEST, 0, just_tested)
-            pop.set_present_variable(col.NP_LAST_TEST, 0, just_tested)
+        Update the test mark of a sub-population based on a given probability.
+        """
+        if len(sub_pop) > 0:
+            # mark people for testing
+            r = rng.uniform(size=len(sub_pop))
+            marked = r < prob_test
+            # set outcomes
+            pop.set_present_variable(col.TEST_MARK, True, sub_pop=pop.apply_bool_mask(marked, sub_pop))
+
+    def test_mark_general_pop(self, pop):
+        """
+        Mark general population to undergo testing this time step.
+        """
+        # testing occurs after a certain year if there is no covid disruption
+        if ((pop.date.year >= self.date_start_testing)
+           & (not (self.covid_disrup_affected | self.testing_disrup_covid))):
+
+            # update testing probabilities
+            self.rate_first_test = self.init_rate_first_test + (min(pop.date.year, self.date_test_rate_plateau)
+                                                                - self.date_start_testing) \
+                                                                * self.an_lin_incr_test
+            self.rate_rep_test = (min(pop.date.year, self.date_test_rate_plateau)
+                                  - self.date_start_testing) * self.an_lin_incr_test
+
+            # general population ready for testing
+            testing_population = pop.get_sub_pop([(col.HARD_REACH, op.eq, False),
+                                                  (col.AGE, op.ge, 15),
+                                                  (col.HIV_STATUS, op.eq, False),
+                                                  [(col.LAST_TEST_DATE, op.le, pop.date -
+                                                    timedelta(days=self.days_to_wait[self.eff_max_freq_testing])),
+                                                   (col.LAST_TEST_DATE, op.eq, None)],
+                                                  (col.TEST_MARK, op.eq, False)])
+
+            if len(testing_population) > 0:
+                # mark people for testing
+                marked = pop.transform_group([col.EVER_TESTED, col.NP_LAST_TEST, col.NSTP_LAST_TEST],
+                                             self.calc_testing_outcomes, sub_pop=testing_population)
+                # set outcomes
+                pop.set_present_variable(col.TEST_MARK, marked, testing_population)
+
+    def calc_testing_outcomes(self, repeat_tester, np_last_test, nstp_last_test, size):
+        """
+        Uses the HIV test probability for either
+        first-time or repeat testers to return testing outcomes.
+        """
+        prob_test = self.calc_prob_test(repeat_tester, np_last_test, nstp_last_test)
```

Comment:
This function `calc_prob_test` doesn't do anything. I think this needs to be fixed?